src/main/java/com/karankumar/bookproject/backend/service/TagService.java

```diff
@@ -0,0 +1,58 @@
+package com.karankumar.bookproject.backend.service;
+
+import java.util.List;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import com.karankumar.bookproject.backend.entity.BaseEntity;
+import com.karankumar.bookproject.backend.entity.Tag;
+import com.karankumar.bookproject.backend.repository.TagRepository;
+
+import lombok.extern.java.Log;
+
+@Service
+@Log
+public class TagService extends BaseService<Tag, Long> {
+
+    private TagRepository tagRepository;
```

Comment:
This variable is never assigned. I recommend setting it to an autowired value in a `TagService` constructor (see the `ReadingGoalService` for an example)